### PR TITLE
Remove lxml dependency

### PR DESCRIPTION
**Pull request by sloretz**
Continues #40 which is a continuation from #32. This removes the dependency on `lxml`.

### CODE REVIEW

File: src/urdf_parser_py/xml_reflection/basics.py
Comment (clalancette) [inline src/urdf_parser_py/xml_reflection/basics.py:14]: I had two major [comments](https://github.com/ros/urdf_parser_py/pull/32#pullrequestreview-135819431) on the original PR.

The first one was about pretty printing the output.  That one isn't super-important, but was something that the old one did.  It looks like this one is doing that as well, which is good, and takes care of that point.

The second point was about preserving XML comments, which lxml does, but minidom does not.  I guess we have to answer whether that is an important use case, and whether we can make that happen with minidom.  I don't know that answer to either one.

Comment (sloretz) [reply src/urdf_parser_py/xml_reflection/basics.py:14]: It looks like the `melodic-devel` branch is explicitly filtering XML comments out. Is there another API in this repo that gives access to them?

https://github.com/ros/urdf_parser_py/blob/aa7199facecd1227409fb9858eb8ccd90fa8716e/src/urdf_parser_py/xml_reflection/basics.py#L34-L39

Comment (clalancette) [reply src/urdf_parser_py/xml_reflection/basics.py:14]: I was originally worried about round-tripping through this API, like:

```
from urdf_parser_py.urdf import URDF

with open('robot.urdf', 'r') as infp:
    robot = URDF.from_xml_string(infp.read())

print(robot.to_xml_string())
```

But you are right, we are already stripping out those comments.  Personally I think that is a bug, but that is orthogonal to this PR.  So I think we can proceed.

File: src/urdf_parser_py/xml_reflection/core.py
Comment (clalancette) [inline src/urdf_parser_py/xml_reflection/core.py:587]: I guess we are inheriting `ET` by importing from `basics.*` at the top, but I think it would be a bit better to explicitly do it at the top of this file:

```
from xml.etree import ElementTree as ET
```

Minor comment, though, I'll leave it to you whether you think we should do that.

**Review by clalancette — APPROVED** [review]

File: src/urdf_parser_py/xml_reflection/core.py
Comment (sloretz) [reply src/urdf_parser_py/xml_reflection/core.py:587]: I removed the `*` import from `core.py` in 0ae72598fda6386db175e4e5d6562522de244541 . `sdf` and `urdf` also `import *`, but I'd like to fix that in a follow up PR.

**Review by clalancette — APPROVED** [review]
I took a quick look at 0ae7259 , just to try to make sure we got all of the imports.  Looks pretty good to me, though we'd need more comprehensive tests to be sure.

Good to merge from me, assuming it passes the tests we already have.

**Pull request merged**
